packages/timeline-state-resolver/src/integrations/tricaster/triCasterStateDiffer.ts

```diff
@@ -300,21 +301,29 @@ export class TriCasterStateDiffer {
 		source: CommandName.CROSSPOINT_SOURCE,
 	}
 
-	private effectCommandGenerator: CommandGeneratorFunction<
+	private keyerEffectCommandGenerator: CommandGeneratorFunction<
 		TriCasterTransitionEffect,
-		RequiredDeep<TriCasterMixEffectState | TriCasterKeyerState>
-	> = ({ value, target, state }) => {
-		const commands: TriCasterCommand[] = []
-		if (value === 'cut') {
-			return commands
-		}
-		if (typeof value === 'number') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value })
-		} else if (value === 'fade') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value: 1 })
+		RequiredDeep<TriCasterKeyerState>
+	> = this.effectCommandGenerator(CommandName.SELECT_INDEX)
+
+	private mixEffectEffectCommandGenerator: CommandGeneratorFunction<
+		TriCasterTransitionEffect,
+		RequiredDeep<TriCasterMixEffectState>
+	> = this.effectCommandGenerator(CommandName.SET_MIX_EFFECT_BIN_INDEX)
+
+	private effectCommandGenerator(
+		selectCommand: TriCasterGenericCommandName<number>
+	): CommandGeneratorFunction<TriCasterTransitionEffect, RequiredDeep<TriCasterKeyerState | TriCasterMixEffectState>> {
+		return ({ value, target, state }) => {
+			if (value === 'cut') {
+				return []
+			}
+			const valueNumber = value === 'fade' ? 0 : value
```

Comment:
What's the type of `valueNumber`? It's seems that it's possible to assign it both a number and a string.

Sidenote: Instead of naming a variable a type i.e. `const valueNumber` you should specify the type instead `const value: number`

Reply:
> It's seems that it's possible to assign it both a number and a string

But it's not because we already filtered out 'cut' and 'fade'

> Instead of naming a variable a type i.e. const valueNumber you should specify the type instead const value: number

That does not apply here. I already have a `value` of different type. This is a case of narrowing the type down (but could as well be a case of converting or parsing the value)

this will be improved a bit in https://github.com/tv2/tv-automation-state-timeline-resolver/pull/50